examples/sync/src/bin/server.rs

```diff
@@ -0,0 +1,438 @@
+            state.inc_errors();
+            break;
+        }
+    }
+
+    Ok(())
+}
+
+fn main() {
+    // Initialize tracing with a clean format
+    tracing_subscriber::fmt()
+        .with_target(false)
+        .with_thread_ids(false)
+        .with_file(false)
+        .with_line_number(false)
+        .init();
+
+    // Parse command line arguments
+    let matches = Command::new("ADB Sync Server")
+        .version("1.0")
+        .about("Serves ADB operations and proofs to sync clients")
+        .arg(
+            Arg::new("port")
+                .short('p')
+                .long("port")
+                .value_name("PORT")
+                .help("Port to listen on")
+                .default_value("8080"),
+        )
+        .arg(
+            Arg::new("initial-ops")
+                .short('i')
+                .long("initial-ops")
+                .value_name("COUNT")
+                .help("Number of initial operations to create")
+                .default_value("100"),
+        )
+        .arg(
+            Arg::new("storage-dir")
+                .short('d')
+                .long("storage-dir")
+                .value_name("PATH")
+                .help("Storage directory for database")
+                .default_value("/tmp/adb_sync_server"),
+        )
+        .get_matches();
+
+    let config = ServerConfig {
+        port: matches
+            .get_one::<String>("port")
+            .unwrap()
+            .parse()
+            .unwrap_or_else(|e| {
+                eprintln!("❌ Invalid port: {e}");
+                std::process::exit(1);
+            }),
+        initial_ops: matches
+            .get_one::<String>("initial-ops")
+            .unwrap()
+            .parse()
+            .unwrap_or_else(|e| {
+                eprintln!("❌ Invalid initial operations count: {e}");
+                std::process::exit(1);
+            }),
+        storage_dir: matches
+            .get_one::<String>("storage-dir")
+            .unwrap()
+            .to_string(),
+    };
+
+    info!("ADB Sync Server starting");
+    info!(
+        port = config.port,
+        initial_ops = config.initial_ops,
+        storage_dir = %config.storage_dir,
+        "Configuration"
+    );
+
+    let executor = tokio_runtime::Runner::default();
+    executor.start(|context| async move {
+        // Create and initialize database
+        let db_id = generate_db_id(&context);
+        let db_config = create_adb_config(&db_id);
+
+        info!(db_id = %db_id, "Initializing database");
+
+        let mut database = match Database::init(context.clone(), db_config).await {
+            Ok(db) => db,
+            Err(e) => {
+                error!(error = %e, "❌ Failed to initialize database");
+                return;
+            }
+        };
+
+        // Create and add initial operations
+        let initial_ops = create_test_operations(config.initial_ops, 12345);
```

Comment:
nit: Let's make this configurable?

Ideally the server just keeps producing continuously (and the client just uses whatever the latest size is to fetch to).

Reply:
> Ideally the server just keeps producing continuously (and the client just uses whatever the latest size is to fetch to).

Created https://github.com/commonwarexyz/monorepo/issues/1254 to track this